### ISSUE TITLE
`pmount` fails with `asprintf: Invalid argument`.

### ISSUE THREAD

**Issue by dreirund** [MisterDA/pmount#3]
Ahoj,

I tried this repository, because `pmount` release version 0.9.23 does not work anymore (fails to mount at least `vfat`, maybe also others, see [&nearr;&hairsp;this topic in the Arch Linux forum](https://bbs.archlinux.org/viewtopic.php?pid=2111421)).

But when I compile your code and want to do a `pmount sdb1`, it fails with  
```
asprintf: Invalid argument
```

Regards!

**Comment by MisterDA** [comment]
Yep yep, I rebased my code at some point and lost track of the changes, so it's broken. I'd need to spend some time re-doing everything!

**Comment by dreirund** [comment]
> Yep yep, I rebased my code at some point and lost track of the
> changes, so it's broken. I'd need to spend some time re-doing
> everything!

OK. The original (and also your latest release version) is also not
working anymore, see the linked forum post.

**Comment by dreirund** [comment]
> I rebased my code at some point and lost track of the changes, so it's broken.

I then want to add to this issue to place a prominent (e.g. at the beginning in bold font) note into the `README.md` stating that current git `master` branch is broken, e.g.:

```
***NOTE:***  
***Currently this code is broken. The [release versions](/MisterDA/pmount/tags) still work, but the git `master` branch will produce non-working code.***
```

Regards!

**Comment by MisterDA** [comment]
Yep yep yep

May I ask for your opinion?
I reformatted the whole code base with class because the code was using inconsistent style, and it's easier to write my own patches.
If I start re-applying my patches as ideas, should I patch pmount first, the reformart? reformat first then patches (what I have done here)? forget about reformatting?

**Comment by dreirund** [comment]
Ahoj,

> *May I ask for your opinion?*  
> *I reformatted the whole code base with class because the code was using inconsistent style, and it's easier to write my own patches.*

Wow great, sounds like you have the aspiration to really take care of it.

> *If I start re-applying my patches as ideas, should I patch pmount first, the reformart? reformat first then patches (what I have done here)? forget about reformatting?*

Actually, I am less a coder than a sysadmin.

But from my point of view, if your reformatting is part of a general cleanup that helps maintaining things later, I from my side would first clean up (thereby making sure that things still work) and then write patches.

Or I might decide not reformat at all, or reformat only parts I touch and do not reformat parts I don't touch anyway.

Regards!

**Comment by vdanjean** [comment]
Hi,

I would have first merged patches that were written with the old formatting, then reformat.
This would probably allow cleaner merges (the patch just applies) and less work.
If you so reformat before, you would need to rewrite/reformat the code of each (old) patch you want to get (and to be sure you do not introduce new bugs when doing so).

Regards
